src/Models/T1_relaxometry/vfa_t1.m

```diff
@@ -93,19 +93,16 @@
             TR = obj.Prot.VFAData.Mat(1,2);
             E = exp(-TR/x.T1);
             Smodel = x.M0*sin(flipAngles/180*pi)*(1-E)./(1-E*cos(flipAngles/180*pi));
-
         end
 
        function FitResult = fit(obj,data)
             % T1 and M0
             flipAngles = (obj.Prot.VFAData.Mat(:,1))';
             TR = obj.Prot.VFAData.Mat(:,2);
-            if ~isfield(data,'B1map'), data.B1map=1; end
-            [m0, t1] = mtv_compute_m0_t1(double(data.VFAData), flipAngles, TR(1), data.B1map);
-            FitResult.T1 = t1;
-            FitResult.M0 = m0;
-
-
+            if (length(unique(TR))~=1), error('VFA data must have same TR'); end
+            if ~isfield(data, 'B1map'), data.B1map = []; end
+            if ~isfield(data, 'Mask'), data.Mask = []; end
+            [FitResult.T1, FitResult.M0] = Compute_M0_T1_OnSPGR(double(data.VFAData), flipAngles, TR(1), data.B1map, data.Mask);
```

Comment:
The mask is not being properly given/handled in this function. You pass it as the fifth parameter, however it is not what is expected by the function (the fifth parameter is the verbosity), and the mask isn't being handled/applied in that function.

https://github.com/qMRLab/qMRLab/blob/27e50f27392e9e1d420ae7b3526e3afac43e38a6/src/Models_Functions/MTV/Compute_M0_T1_OnSPGR.m#L1

Reply:
I just realised that my comment is completely wrong; it works fine and as intended. My migraine must have been giving me bursts of dyslexia - sorry about that.